src/LinAlg/hiopMatrixSparseTriplet.cpp

```diff
@@ -539,8 +550,13 @@ hiopMatrix* hiopMatrixSymSparseTriplet::new_copy() const
   return copy;
 }
 
-/* block of W += alpha*this 
- * Note W; contains only the upper triangular entries */
+/** 
+ * @brief block of W += alpha*this 
+ * @note W contains only the upper triangular entries
+ * 
+ * @warning This method should not be called directly.
+ * Use addUpperTriangleToSymDenseMatrixUpperTriangle instead.
+ */
 void hiopMatrixSymSparseTriplet::addToSymDenseMatrixUpperTriangle(int row_start, int col_start, 
```

Comment:
function `addToSymDenseMatrixUpperTriangle` has never been used in the code. 
This kind of functions, i.e, `addToSymDenseMatrixUpperTriangle` and `transAddToSymDenseMatrixUpperTriangle` are used to build KKT matrix, and hence only `transAddToSymDenseMatrixUpperTriangle` is required in the current code.
Maybe we should remove it from the base class [see here](https://github.com/LLNL/hiop/blob/2252b90bc40f86b8d842da6fec30105feddbf647/src/LinAlg/hiopMatrix.hpp#L140) 
Tested with Ex1,2,4,5, and all of them can run without this function.

Reply:
I think currently `addUpperTriangleToSymDenseMatrixUpperTriangle` is implemented to call `addToSymDenseMatrixUpperTriangle`. Perhaps it would be a good idea to create a separate GitHub issue to discuss how this is best handled? 

When porting to Umpire/RAJA we tried to make minimal changes to HiOp code.

Reply:
see a4022c8 and comments in #98